Remotion/Validation/Core/Results/ObjectValidationFailure.cs

```diff
@@ -15,18 +15,36 @@
 // along with re-motion; if not, see http://www.gnu.org/licenses.
 // 
 using System;
+using System.Collections.Generic;
 using JetBrains.Annotations;
+using Remotion.Reflection;
+using Remotion.Utilities;
 
 namespace Remotion.Validation.Results
 {
   public class ObjectValidationFailure : ValidationFailure
   {
+    [NotNull]
+    public IReadOnlyCollection<ValidatedObjectProperty> ValidatedProperties { get; }
```

Comment:
Hmm... Given the naming and usage, is it a good idea to have a set of `ValidatedProperties` where the `Property` can be null? I suggest we accept that property is also mandatory on the DTO and rename the DTO to `ValidatedProperty` which makes the `Object` just the instance that holds the property value. 

Good idea? Bad idea?

Reply:
decision: Use `ValidatedProperty` and make `Property` mandatory